cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1667,10 +1722,14 @@ public List<ConfigMap> generatePerBrokerConfigurationConfigMaps(MetricsAndLoggin
                 // environment variables are parsed in the container bash scripts
                 data.put(BROKER_LISTENERS_FILENAME, listeners.stream().map(ListenersUtils::envVarIdentifier).collect(Collectors.joining(" ")));
 
-                if (useKRaft) {
+                // controller and broker gets the Cluster ID in different states if during migration
+                // anyway they both get it when in full KRaft-mode
+                if ((node.controller() && this.kafkaMetadataConfigState.ordinal() >= KafkaMetadataStateManager.KafkaMetadataConfigurationState.PRE_MIGRATION.ordinal()) ||
+                        (node.broker() && this.kafkaMetadataConfigState.ordinal() >= KafkaMetadataStateManager.KafkaMetadataConfigurationState.MIGRATION.ordinal())) {
```

Comment:
The new line here seems weird. If nothing else, this is how it is done in our other code I think.

```suggestion
                if ((node.controller() && this.kafkaMetadataConfigState.ordinal() >= KafkaMetadataStateManager.KafkaMetadataConfigurationState.PRE_MIGRATION.ordinal())
                        || (node.broker() && this.kafkaMetadataConfigState.ordinal() >= KafkaMetadataStateManager.KafkaMetadataConfigurationState.MIGRATION.ordinal())) {
```